client/lib/media/utils.js

```diff
@@ -232,6 +233,28 @@ var MediaUtils = {
 	},
 
 	/**
+	 * Returns true if the specified item is a valid file in a Premium plan,
+	 * or false otherwise.
+	 *
+	 * @param  {Object}  item Media object
+	 * @param  {Object}  site Site object
+	 * @return {Boolean}      Whether the Premium plan supports the item
+	 */
+	isSupportedFileTypeInPremium: function( item, site ) {
+		if ( ! site || ! item ) {
+			return false;
+		}
+
+		if ( ! MediaUtils.isSiteAllowedFileTypesToBeTrusted( site ) ) {
+			return true;
+		}
+
+		return VideoPressFileTypes.some( function( allowed ) {
+			return allowed.toLowerCase() === item.extension.toLowerCase();
+		} );
```

Comment:
This function only returns true for video files, but the name and description indicate it should return true for all supported files in the premium plan, including images and pdfs.